### PR TITLE
Infrastructure cleanups

### PR DESCRIPTION
**Pull request by sadlerap**
This fixes a few problems I've noticed with how some of our infrastructure around the controller is defined.

### CODE REVIEW

**Review by filariow — APPROVED** [review]

**Comment by mmulholla** [comment]
I ran this an it failed on my mach - it needs USER to be set too

File: test/acceptance/features/steps/kind.py [hunk 1]
```diff
@@ -115,7 +115,9 @@ def __deploy_primaza(self, kubeconfig_path: str, img: str):
         assert err == 0, f"error deploying Primaza's controller into cluster {self.cluster_name}"
 
     def __build_install_primaza_base_cmd(self, kubeconfig_path: str, img: str) -> Command:
+        # if HOME isn't set, intentionally fail
         return Command() \
+            .setenv("HOME", os.getenv("HOME")) \
```
Comment (mmulholla) [inline test/acceptance/features/steps/kind.py:120]: At least for mac this need USER adding too.

**Review by mmulholla — CHANGES_REQUESTED** [review]

File: test/acceptance/features/steps/kind.py (hunk 1 above)
Comment (sadlerap) [reply test/acceptance/features/steps/kind.py:127]: Fixed; try it again and let me know if it works.

Comment (mmulholla) [reply test/acceptance/features/steps/kind.py:127]: Can confirm. Thanks.

**Review by mmulholla — DISMISSED** [review]

**Review by filariow — DISMISSED** [review]

**Comment by sadlerap** [comment]
rebased, please re-approve.

File: make/agents/app/build.mk [hunk 2]
```diff
@@ -15,7 +15,7 @@ run: fmt vet ## Run a controller from your host.
 # (i.e. docker build --platform linux/arm64 ). However, you must enable docker buildKit for it.
 # More info: https://docs.docker.com/develop/develop-images/build_enhancements/
 .PHONY: docker-build
-docker-build: test ## Build docker image with the manager.
```
Comment (filariow) [inline make/agents/app/build.mk:18]: nice, as part of this change I think we can also remove [Go Setup from Push to ghcr workflow](https://github.com/primaza/primaza/blob/da77304464cc2d48d350723121e63ff8ea09fa1d/.github/workflows/push-ghcr.yaml#L21)

**Review by filariow — APPROVED** [review]
lgtm, just one proposal

File: make/agents/app/build.mk (hunk 2 above)
Comment (sadlerap) [reply make/agents/app/build.mk:18]: Rebased and added.

**Review by filariow — APPROVED** [review]
LGTM

**Pull request merged**